File: java/find-missing-and-repeated-values.java

import java.util.HashSet;

class Solution {
    public int[] findMissingAndRepeatedValues(int[][] grid) {
        int n = grid.length;
        HashSet<Integer> seen = new HashSet<>();
        int repeated = -1, missing = -1;
        for (int[] arr : grid) {
            for (int val : arr) {
                if (seen.contains(val)) {
                    repeated = val; 
                } else {
                    seen.add(val);
                }
            }
        }
        for (int i = 1; i <= n * n; i++) {
            if (!seen.contains(i)) {
                missing = i;
                break;
            }
        }
        return new int[]{repeated, missing};
    }
}

File: cpp/find-missing-and-repeated-values.cpp
class Solution {
public:
    vector<int> findMissingAndRepeatedValues(vector<vector<int>>& grid) {
        int xOR=0;
        vector<int> mp(grid.size()*grid.size()+1,0);
        int repeated;
        int i=0;
        for (auto &vec:grid) {
            for (auto &val:vec) {
                i++;
                xOR = xOR ^ val ^ i;
                if (mp[val]!=0) {
                    repeated = val;
                }
                else {
                    mp[val]++;
                }
            }
        }
        return {repeated,xOR^repeated};
    }
};
